### PR TITLE
PubMatic to support userId sub-modules

### PR DESCRIPTION
**Pull request by pm-harshad-mane**
## Type of change

- [X] Feature

## Description of change
Changes are made to support the following userId submodules in PubMatic adapter
+ PubCommon
+ DigiTrust
+ CriteoRTUS
+ ID5ID
+ IdentityLink

### CODE REVIEW

File: modules/pubmaticBidAdapter.js [hunk 1]
```diff
@@ -624,10 +624,33 @@ function _handleTTDId(eids, validBidRequests) {
...
 function _handleEids(payload, validBidRequests) {
   let eids = [];
   _handleDigitrustId(eids);
   _handleTTDId(eids, validBidRequests);
+  const bidRequest = validBidRequests[0];
+  if (bidRequest && bidRequest.userId) {
+    _addExternalUserId(eids, utils.deepAccess(bidRequest, `userId.pubcid`), 'pubcommon', 1);
+    _addExternalUserId(eids, utils.deepAccess(bidRequest, `userId.digitrustid.data.id`), 'digitru.st', 1);
+    _addExternalUserId(eids, utils.deepAccess(bidRequest, `userId.id5id`), 'id5-sync.com', 1);
+    _addExternalUserId(eids, utils.deepAccess(bidRequest, `userId.criteortus.${BIDDER_CODE}.userid`), 'criteortus', 1);
```
Comment (pm-harshad-mane) [inline modules/pubmaticBidAdapter.js]: Hello @jaiminpanchal27 ,
What should be source value for CriteoRTUS?

File: modules/pubmaticBidAdapter.js [hunk 2]
```diff
@@ -624,10 +624,33 @@ function _handleTTDId(eids, validBidRequests) {
...
+    _addExternalUserId(eids, utils.deepAccess(bidRequest, `userId.idl_env`), 'identity_link_envelope', 1);
```
Comment (pm-harshad-mane) [inline modules/pubmaticBidAdapter.js]: Hello @mamatic ,
What should be source value for IdentityLinkIdSystem for ORTB3.0 eids spec?

Comment (pm-harshad-mane) [reply modules/pubmaticBidAdapter.js]: Resolved, setting it as "liveramp.com"

**Comment by bretg** [comment]
Docs PR https://github.com/prebid/prebid.github.io/pull/1472

File: modules/pubmaticBidAdapter.js (hunk 1 above)
Comment (pm-harshad-mane) [reply modules/pubmaticBidAdapter.js]: we can keep source param for CtriteoRTUS as "criteortus", we will change it if needed.

**Comment by pm-harshad-mane** [comment]
Hello @idettman ,
Could you please refer https://github.com/prebid/prebid.github.io/pull/1508 for documentation?

**Pull request merged**